eliHenderson/assignments/fastSlow/lc876.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Summary: 
+ *  If there is no cycle in the linked list 
+ *  there will be a null at the of list
+ *  we can use two pointer to iterate through the 
+ *  array. One pointer will move 2x faster than the 
+ *  other, Once the fast pointer reaches the end, 
+ *  the slower pointer will be a the middle
+ *  return the slow pointer
+ * 
+ * while (fast is not null)
+ *  fast = fast.next.next
+ *  slow = slow.next
+ * 
+ * return slow
+ * 
+ * Time: (O(N))
+ * Space: O(1)
+
+ */
+
+class Node {
+  constructor(value, next = null) {
+    this.value = value;
+    this.next = next;
+  }
+}
+
+const find_middle_of_linked_list = function (head) {
+  let hare = head;
+  let turtle = head;
+  while (hare !== null) {
+    hare = hare.next;
+    if (hare === null) break;
+    hare = hare.next;
+
+    turtle = turtle.next;
+  }
+
+  return turtle;
+};
+
+const find_middle_of_linked_list = function (head) {
+  let hare = head;
+  let turtle = head;
+
+  while (hare !== null) {
+    hare = hare.next;
+    if (hare === null) break;
+    hare = hare.next;
+
+    turtle = turtle.next;
+  }
+
+  return turtle;
+};
+
+const find_middle_of_linked_list2 = (head) => {
+  let hare = head;
+  let turtle = head;
+
+  while (hare !== null) {
+    hare = hare.next.next;
+    turtle = turtle.next;
+  }
+
+  return turtle;
+};
+
+const find_middle_of_linked_list2 = (head) => {
+  let hare = head;
+  let turtle = head;
+
+  while (hare !== null) {
+    hare = hare.next.next;
+    turtle = turtle.next;
+  }
+};
+
```

Comment:
Have you tested this code? I do not believe it will run properly, as there is a chance `hare.next === null` and then `hare.next.next` will produce an error.